camel/functions/search_functions.py

```diff
@@ -72,75 +67,77 @@ def search_duckduckgo(
 
     Args:
         query (str): The query to be searched.
-        num_result_pages (int): The number of result pages to retrieve.
+        source (str): The type of information to query (e.g., "text",
+            "images", "videos"). Defaults to "text".
+        max_results (int): Max number of results, defaults to `10`.
 
     Returns:
         List[Dict[str, Any]]: A list of dictionaries where each dictionary
-        represents a search result.
-            Each dictionary contains the following keys:
-            - 'result_id': A sequential number starting from 1.
-            - 'title': The title or main text of the search result.
-            - 'description': A brief description of the result, often similar
-                to the title.
-            - 'long_description': More detail of the result, often the same as
-                the description.
-            - 'url': The URL of the result.
-
-            Example:
-            {
-                'result_id': 1,
-                'title': 'DuckDuckGo',
-                'description': 'DuckDuckGo is an Internet search engine ' + \
-                    'that emphasizes protecting searchers' privacy.',
-                'long_description': 'DuckDuckGo is an Internet search ' + \
-                    'engine that emphasizes protecting searchers' privacy.',
-                'url': 'https://www.duckduckgo.com'
-            }
-        title, description, url of a search result.
+            represents a search result.
     """
     from duckduckgo_search import DDGS
     from requests.exceptions import RequestException
 
     ddgs = DDGS()
-    responses = []
+    responses: List[Dict[str, Any]] = []
 
-    try:
-        results = ddgs.text(keywords=query, num_result_pages=num_result_pages)
+    if source == "text":
```

Comment:
Hey @jjyaoao , thanks for the review, yes indeed the code is similar here, but if not using `if` statement checking the type of source, we will get all the text, image, video information retrieved, then do the split to pick out the content suit for source type, it will be 3x resource consuming and slower

Reply:
> Hey @jjyaoao , thanks for the review, yes indeed the code is similar here, but if not using `if` statement checking the type of source, we will get all the text, image, video information retrieved, then do the split to pick out the content suit for source type, it will be 3x resource consuming and slower

Thanks for the clarification, I understand your point. However, is there a possibility that the source parameter might receive values other than "text", "images", or "videos"? Should we implement an exception handling mechanism to avoid errors if the upper layer passes in unexpected values?

Reply:
This function would be used by agent, if add exception error for other source type would stop the running of agent. Currently if the source type not supported it will return an empty list

Reply:
Thank you, I understand.